application/src/main/data/upgrade/3.6.3/schema_update.sql

```diff
@@ -14,6 +14,105 @@
 -- limitations under the License.
 --
 
+-- UPDATE PUBLIC CUSTOMERS START
+
+ALTER TABLE customer ADD COLUMN IF NOT EXISTS is_public boolean DEFAULT false;
+UPDATE customer SET is_public = true WHERE title = 'Public';
+
+-- UPDATE PUBLIC CUSTOMERS END
+
+-- UPDATE CUSTOMERS WITH SAME TITLE START
+
+CREATE OR REPLACE PROCEDURE update_customers_with_the_same_title()
+    LANGUAGE plpgsql
+AS
+$$
+DECLARE
+    customer_record  RECORD;
+    dashboard_record RECORD;
+    title_exists     BOOLEAN;
+    new_title        TEXT;
+    updated_json     JSONB;
+BEGIN
+    RAISE NOTICE 'Starting the customer and dashboard update process.';
+
+    FOR customer_record IN
+        SELECT id, tenant_id, title, duplicate_number
+        FROM (
+                 SELECT
+                     id,
+                     tenant_id,
+                     title,
+                     ROW_NUMBER() OVER(PARTITION BY tenant_id, title ORDER BY id) AS duplicate_number
+                 FROM customer
+             ) AS duplicate_customers
+        WHERE duplicate_number > 1
+        LOOP
+            -- Attempt with 'duplicate' suffix
+            new_title := customer_record.title || ' duplicate ' || (customer_record.duplicate_number - 1)::TEXT;
+
+            -- Check if new_title already exists for the same tenant_id
+            SELECT EXISTS (
+                SELECT 1
+                FROM customer
+                WHERE tenant_id = customer_record.tenant_id
+                  AND title = new_title
+            ) INTO title_exists;
+
+            -- If generated title exists, use customer id instead to create a unique title
+            IF title_exists THEN
+                new_title := customer_record.title || ' duplicate ' || customer_record.id::TEXT;
```

Comment:
Since we are running the update, no one can change the table, so it is safe to do this in Java part. You may also be sure that you may use 
'Customer A ' 
'Customer A 2' 
'Customer A 3' 
...
'Customer A n'